src/pages/staticPages/tools/oncoprinter/Oncoprinter.tsx

```diff
@@ -379,9 +397,177 @@ export default class Oncoprinter extends React.Component<
         return WindowStore.size.width - 75;
     }
 
+    @action.bound
+    public handleSelectedClinicalTrackColorChange(
+        value: string,
+        color: RGBAColor | undefined
+    ) {
+        if (this.selectedClinicalTrack) {
+            this.props.store.setUserSelectedClinicalTrackColor(
+                this.selectedClinicalTrack.label,
+                value,
+                color
+            );
+        }
+    }
+
+    @observable trackKeySelectedForEdit: string | null = null;
+
+    @action.bound
+    setTrackKeySelectedForEdit(key: string | null) {
+        this.trackKeySelectedForEdit = key;
+    }
+
+    @computed get selectedClinicalTrack() {
+        return _.find(
+            this.props.store.clinicalTracks,
+            t => t.key === this.trackKeySelectedForEdit
+        );
+    }
+
+    @computed get defaultClinicalAttributeColoringForStringDatatype() {
+        if (this.selectedClinicalTrack) {
+            return _.mapValues(
+                getDefaultClinicalAttributeColoringForStringDatatype(
+                    this.selectedClinicalTrack.data
+                ),
+                hexToRGBA
+            );
+        }
+        return _.mapValues(RESERVED_CLINICAL_VALUE_COLORS, hexToRGBA);
+    }
+
+    @autobind
+    private getDefaultSelectedClinicalTrackColor(value: string) {
+        if (!this.selectedClinicalTrack) {
+            return [255, 255, 255, 1];
+        }
+        if (this.selectedClinicalTrack.datatype === 'counts') {
+            return MUTATION_SPECTRUM_FILLS[
+                _.indexOf(MUTATION_SPECTRUM_CATEGORIES, value)
+            ];
+        } else if (this.selectedClinicalTrack.datatype === 'string') {
+            // Mixed refers to when an event has multiple values (i.e. Sample Type for a patient event may have both Primary and Recurrence values)
+            if (value === 'Mixed') {
+                return [220, 57, 18, 1];
+            } else {
+                return this.defaultClinicalAttributeColoringForStringDatatype[
+                    value
+                ];
+            }
+        }
+        return [255, 255, 255, 1];
+    }
+
+    @computed get selectedClinicalTrackValues() {
+        if (this.selectedClinicalTrack) {
+            return getClinicalTrackValues(this.selectedClinicalTrack);
+        }
+        return [];
+    }
+
+    @observable clinicalTrackColorChanged: boolean = false;
+
+    @action.bound
+    setClinicalTrackColorChanged(changed: boolean) {
+        this.clinicalTrackColorChanged = changed;
```

Comment:
seems odd to keep track of changed this way.  i would expect this status to be derivable from the selection itself?

Reply:
the track changed can be found by checking if the value-to-color mappings are equal before and after change. i thought it would be better to have a property we can check that just tells us if something has changed in the mapping rather than doing an equality check between the two objects every time.